init.go

```diff
@@ -0,0 +1,42 @@
+package codacytool
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"time"
+)
+
+var (
+	sourceDirFlag           *string
+	toolConfigsBasePathFlag *string
+)
+
+const (
+	defaultTimeout = 15 * time.Minute
+)
+
+// StartTool receives the tool implementation as parameter and run the tool
+func StartTool(impl ToolImplementation) {
+	parseFlags()
+
+	os.Setenv("TOOL_CONFIGS_BASEPATH", *toolConfigsBasePathFlag)
```

Comment:
why are we setting an env variable instead of passing it as an argument to the receiving functions?

Reply:
Indeed it does not make sense to set environment variables in runtime.
Instead of using env variable after, I used the `toolConfigsBasePathFlag` directly.